org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/IContentProvider.java

```diff
@@ -0,0 +1,43 @@
+/*******************************************************************************
+ * Copyright (c) 2017 David Gileadi and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     David Gileadi - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.ls.core.internal;
+
+import java.net.URI;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.ls.core.internal.preferences.Preferences;
+
+/**
+ * Interface for classes that provide text content from resources.
+ *
+ * @see IDecompiler
+ */
+public interface IContentProvider {
+	/**
+	 * Accept preferences which may contain configuration for this content provider.
+	 *
+	 * @param preferences
+	 */
+	void setPreferences(Preferences preferences);

```

Comment:
`default void setPreferences(Preferences preferences){}` so you don't need to implement empty methods